File: src/main/java/decorator/VipDiscount.java
package decorator;

public class VipDiscount implements Discount{
    public static final double RATE = 0.05;
    public static final String HISTORY = "VIP 할인";

    @Override
    public double amountAfterDiscount(double amount) {
        return amount * RATE;
    }

    @Override
    public String history() {
        return HISTORY;
    }
}

File: src/main/kotlin/decorator/VipDiscount.kt
package decorator

class VipDiscount: Discount {
    companion object {
        const val RATE = 0.05
        const val HISTORY = "VIP 할이"
    }

    override fun amountAfterDiscount(amount: Double): Double {
        return amount * RATE
    }

    override fun history(): String {
        return HISTORY
    }
}